jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/applications/DeleteApplicationActionTest.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.applications
+
+import assertk.assertions.hasMessageContaining
+import assertk.assertions.isInstanceOf
+import com.intellij.testFramework.ProjectRule
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.awscore.exception.AwsErrorDetails
+import software.amazon.awssdk.services.cloudformation.CloudFormationClient
+import software.amazon.awssdk.services.cloudformation.model.CloudFormationException
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksRequest
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksResponse
+import software.amazon.awssdk.services.cloudformation.model.Stack
+import software.amazon.awssdk.services.cloudformation.model.StackStatus
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.jetbrains.utils.delegateMock
+import java.util.concurrent.TimeoutException
+
+class DeleteApplicationActionTest {
```

Comment:
How are we testing the 'waiter' makes multiple calls ?

Reply:
Reworked the testing. Adding call times verification.